c7n/commands.py

```diff
@@ -125,6 +125,12 @@ def run(options, policies):
 def report(options, policies):
     assert len(policies) == 1, "Only one policy report at a time"
     policy = policies.pop()
+    odir = options.output_dir
+    if os.path.sep in odir and os.path.basename(odir) == policy.name:
+        # policy sub-directory passed - ignore
+        options.output_dir = os.path.split(odir)[0]
```

Comment:
on a windows client pointing to s3 output location this would do the wrong thing, i think you'll need to detect if odir.startswith('s3://') to do the right thing. alternatively for the initial condition odir.endswith(policy.name) ... you'll also have to handle trailing slashes on the output dir.